crates/iota-graphql-rpc/src/types/transaction_block/mod.rs

```diff
@@ -231,6 +232,21 @@ impl TransactionBlock {
             TransactionBlockInner::DryRun { .. } => None,
         }
     }
+
+    /// Returns whether the transaction has been indexed on the fullnode.
+    async fn indexed_on_node(&self, ctx: &Context<'_>) -> Result<Option<bool>> {
+        let fullnode_client = get_fullnode_client(ctx)?;
+        let Some(digest) = self.native_signed_data().map(|d| d.digest()) else {
+            // dry-run transactions are never indexed
+            return Ok(Some(false));
+        };
+        Ok(Some(
+            fullnode_client
+                .http()
+                .is_transaction_indexed_on_node(digest)
```

Comment:
I'm thinking a bit about performance here.
Is it possible that someone will request this field for a page of transactions, resulting in a separate request to fullnode for every item on the page?

Is this a consideration? Maybe we should look at some optimizations. (single request for many digests? or maybe we could keep in memory checkpoint id of last tx indexed for given fullnode, so we know that everything that happened earlier is already indexed, without hitting the fullnode, for querying historical data this would practically mean we don't have to query the fullnode at all)